packages/core-data/src/utils/test/if-matching-action.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+import ifMatchingAction from '../if-matching-action';
+
+describe( 'ifMatchingAction', () => {
+	function createEnhancedReducer( predicate ) {
+		const enhanceReducer = ifMatchingAction( predicate );
+		return enhanceReducer( () => 'Called' );
+	}
+
+	it( 'should call reducer if predicate returns true', () => {
+		const reducer = createEnhancedReducer( () => true );
```

Comment:
Isn't it edge case when state is `undefined`? There should be another test for an existing state.

Reply:
I guess it's not important because the idea is just to check that the reducer was called or not, I'm updating the default state anyway :)